android/src/main/java/com/appfolio/uploader/ModifiedHttpUploadRequest.kt

```diff
@@ -49,4 +49,8 @@ abstract class ModifiedHttpUploadRequest<B : HttpUploadRequest<B>>(context: Cont
 
     return uploadTaskParameters.id;
   }
+
+  fun setCustomUploadID(uploadID: String) {
+    this.uploadId = uploadID
```

Comment:
Also call setUploadId so it updates the property in the base class in case that is being used elsewhere 

```suggestion
    this.uploadId = uploadID
    setUploadID(uploadID)
```